pkg/entity/register/worker.go

```diff
@@ -60,47 +73,97 @@ func (w *worker) Run(ctx context.Context) {
 
 			if batchSize == w.maxBatchSize {
 				timer.Reset(w.maxBatchDuration)
-				w.send(batch, &batchSize)
+				w.send(ctx, batch, &batchSize)
 			}
 
 		case <-timer.C:
 			if len(batch) > 0 {
-				w.send(batch, &batchSize)
+				w.send(ctx, batch, &batchSize)
 			}
 			timer.Reset(w.maxBatchDuration)
 		}
 	}
 }
 
-func (w *worker) send(batch map[entity.Key]fwrequest.EntityFwRequest, batchSize *int) {
+func (w *worker) send(ctx context.Context, batch map[entity.Key]fwrequest.EntityFwRequest, batchSize *int) {
 	defer w.resetBatch(batch, batchSize)
 
 	var entities []entity.Fields
 	for _, r := range batch {
 		entities = append(entities, r.Data.Entity)
 	}
-	responses, _, errClient := w.client.RegisterBatchEntities(w.agentIDProvide().ID, entities)
-	if errClient != nil {
-		// TODO error handling
-		return
-	}
+
+	responses := w.registerEntitiesWithRetry(ctx, entities)
 
 	for _, resp := range responses {
-		if resp.Err != "" {
-			// TODO error handling
+		if resp.ErrorMsg != "" {
+			wlog.WithError(fmt.Errorf(resp.ErrorMsg)).
+				WithField("entityName", resp.Name).
+				Errorf("failed to register entity")
 			continue
 		}
 
-		r, ok := batch[resp.Key]
+		if len(resp.Warnings) > 0 {
+			for _, warn := range resp.Warnings {
+				wlog.
+					WithField("entityName", resp.Name).
+					WithField("entityID", resp.ID).
+					Errorf("entity registered with warnings: %s", warn)
+			}
+		}
+
+		r, ok := batch[entity.Key(resp.Name)]
 		if !ok {
-			// TODO error handling
+			wlog.
+				WithField("entityName", resp.Name).
+				WithField("entityID", resp.ID).
+				WithField("entityName", resp.Name).
+				Errorf("entityName returned by register not found in the entities batch")
+			continue
 		} else {
 			r.RegisteredWith(resp.ID)
 			w.reqsRegisteredQueue <- r
 		}
 	}
 }
 
+// registerEntitiesWithRetry will submit entities to the backend for registration.
+// In case of StatusCodeConFailure or StatusCodeLimitExceed errors it will retry with backoff.
+// for other errors, data will be discarded.
+func (w *worker) registerEntitiesWithRetry(ctx context.Context, entities []entity.Fields) []identityapi.RegisterEntityResponse {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+
+		var err error
+		responses, err := w.client.RegisterBatchEntities(w.agentIDProvide().ID, entities)
+		if err == nil {
+			w.retryBo.Reset()
```

Comment:
Take into account that we are backing off just a worker, not the rest of the submissions, which I guess might be expected behaviour.

Anyway, this behaviour might be accepted, could you reach infra platform to confirm?